helpers/telemetryHelper.js

```diff
@@ -0,0 +1,145 @@
+const Telemetry = require('sb_telemetry_util')
+const telemetry = new Telemetry()
+const _ = require('lodash')
+const fs = require('fs')
+const path = require('path');
+const pdata = { id: 'discussion-middleware', pid: '', ver: '1.0.0' }
+
+const telemtryEventConfig = JSON.parse(fs.readFileSync(path.join(__dirname, './telemetryEventConfig.json')))
+
+
+module.exports = {
+
+  /**
+   * @description - To log the API Error event
+   * @param  {} req
+   * @param  {} res
+   * @param  {} data
+   * @param  {} proxyResData
+   * @param  {} proxyRes
+   * @param  {} resCode
+   * @param  {} env
+   */
+  logTelemetryErrorEvent: function (req, res, data, proxyResData, proxyRes, resCode, env) {
+    let telemetryObj = this.getTelemetryObject(proxyRes, data, proxyResData, resCode);
+    const option = telemetry.getTelemetryAPIError(telemetryObj, proxyRes, env);
+    if (option) {
+      let object = telemetryObj.obj || {}
+      let channel = _.get(req.headers, 'x-channel-id') || ''
+      const context = {
+        channel: channel,
+        env: env,
+        cdata: [],
+        pdata: pdata,
+        did: _.get(req.headers, 'x-device-id') || '',
+        sid: _.get(req.headers, 'x-session-id') || ''
+      }
+      context.pdata.pid = _.get(req.headers, 'x-app-id') || '';
+      const actor = {
+        id: req.headers['x-consumer-id'] || telemtryEventConfig.default_userid,
+        type: req.headers['x-consumer-username'] || telemtryEventConfig.default_username
+      }
+      telemetry.error({
+        edata: option.edata,
+        context: _.pickBy(context, value => !_.isEmpty(value)),
+        object: _.pickBy(object, value => !_.isEmpty(value)),
+        actor: _.pickBy(actor, value => !_.isEmpty(value))
+      })
+    }
+  },
+
+  /**
+   * This function helps to log API entry event
+   */
+  logAPIEvent: function (req, uri) {
+    const apiConfig = telemtryEventConfig.URL[uri] || {}
+    if (req) {
+      params = this.getParamsData(req, '', {}, uri)
+    }
+    const edata = telemetry.logEventData('api_call', 'TRACE', apiConfig.message, JSON.stringify(params))
+    edata.message = JSON.stringify({ title: 'API Log', url: req.path })
+    let object = {}
+    let channel = _.get(req.headers, 'x-channel-id') || ''
+    const context = {
```

Comment:
Adding context to any event should be generic. Because most of the values will be common for all the events.

I can see the context is added again in the above function logTelemetryErrorEvent